pkg/receive/handler.go

```diff
@@ -208,6 +209,15 @@ func NewHandler(logger log.Logger, o *Options) *Handler {
 				Buckets:   []float64{10, 50, 100, 500, 1000, 5000, 10000},
 			}, []string{"code", "tenant"},
 		),
+		writeE2eLatency: promauto.With(registerer).NewHistogramVec(
+			prometheus.HistogramOpts{
+				Namespace: "thanos",
+				Subsystem: "receive",
+				Name:      "write_e2e_latency_seconds",
+				Help:      "The end-to-end latency of the oldest sample in write requests.",
+				Buckets:   []float64{1, 5, 10, 20, 30, 40, 50, 60, 90, 120, 300, 600, 900, 1200, 1800, 3600},
```

Comment:
cardinality estiamte: 16 le x 5 code x 20 tenants x 30 pods = 48k seems fine.

Reply:
i'm thinking if we can drop "code" label as failed requests can be captured by other metrics. but even if we do not drop it, the data samples will be minimal. WDYT?

Reply:
why do we need latency buckets like 3600?

Reply:
In normal cases e2e latency can be ~ 10 minutes. Extending the bucket to 1 hour to (hopefully) cover future outage